spirv_glsl.cpp

```diff
@@ -17712,7 +17712,7 @@ void CompilerGLSL::emit_block_chain(SPIRBlock &block)
 
 		if (!collapsed_switch)
 		{
-			if (block_like_switch && !is_legacy_es())
+			if ((block_like_switch || is_legacy()) && !is_legacy_es())
```

Comment:
This is weird. Can you explain why this isn't just `if (block_like_switch && !is_legacy())`?

Reply:
In legacy ES, do/while loops are not supported, so a for loop is inserted around the if statements. In legacy desktop, a do/while loop is inserted. Therefore, if the version is a legacy one AND not ES, a `while(false)` needs to be inserted.